File: Spring project/front-end/src/services/CategoryService.js
import axios from "axios";

const CATEGORY_REST_API_BASE_URL = "http://localhost:8080/api/categories";

export const getAllCategories = () => axios.get(CATEGORY_REST_API_BASE_URL);

export const createCategory = (category) =>
  axios.post(CATEGORY_REST_API_BASE_URL, category);

export const getCategoryById = (categoryId) =>
  axios.get(CATEGORY_REST_API_BASE_URL + "/" + categoryId);

export const updateCategory = (categoryId, category) =>
  axios.put(CATEGORY_REST_API_BASE_URL + "/" + categoryId, category);

export const deleteCategory = (categoryId) =>
  axios.delete(CATEGORY_REST_API_BASE_URL + "/" + categoryId);

File: Spring project/back-end/java-springboot-project/src/main/java/lt/techin/java_springboot_project/service/CategoryService.java
package lt.techin.java_springboot_project.service;

import lt.techin.java_springboot_project.dto.CategoryDto;

import java.util.List;

public interface CategoryService {
    CategoryDto createCategory(CategoryDto categoryDto);

    CategoryDto getCategoryById(Long categoryId);

    List<CategoryDto> getAllCategories();

    CategoryDto updateCategory(Long categoryId, CategoryDto updatedCategory);

    void deleteCategory(Long categoryId);
}
